R/DIFFUSE.R

```diff
@@ -0,0 +1,266 @@
+                             AGGedges$to[edgeIndex])
+            #reduce to unique elements
+            geneList <- unique(geneList)
+            #remove vertex of interest from list
+            geneList <- geneList[-grep(HGNCsymb[i],geneList, fixed = TRUE)]
+
+            deg <- length(geneList) #degree of vertex of interest
+
+            vertDegree[grep(HGNCsymb[i],names(vertDegree), fixed = TRUE)] <- deg
+
+        }
+
+    #Assigning values to matrix W: See Leiseron et. al 2015 for procedure description
+
+        consoleVect <-  "Constructing matrix W"
+        if (!silent) {
+            print(consoleVect)
+        }
+
+        doneEdges <- 0
+
+        for (n in 1:sizeVerts) {
+
+
+
+           jElement <- AGGverts$name[n]
+
+           #look for directional edges with gene j as the source
+           jEdgeIndex <- as.logical(grepl(jElement, AGGedges$from, fixed = TRUE))
+
+           #List of receivers of directed edge from j
+           jTargetList <- AGGedges$to[jEdgeIndex]
+
+           for (m in 1:length(jTargetList)) {
+               iElement <- jTargetList[m] #retrieve name of target of node j
+
+               #W [i (receiver) , j (source)] is equal to 1/degree(j)
+               W[iElement,jElement] <- 1/vertDegree[names(vertDegree) == jElement]
+
+               doneEdges <- doneEdges + 1
+
+               consoleVect <- paste("Have value 1/deg(j) for",
+                                                   doneEdges,
+                                                   "of", sizeEdges,
+                                                   "assigned to W", sep = " ")
+               if (!silent) {
+                   print(consoleVect)
+               }
+
+           }
+
+        } #end construction of W
+
+    #======== Part 3 of Leis: calculating the matrix F ==============
+        betaVal <- param[[1]] #set beta Value, or the 'insulation' coefficient
+
+        #Following equation F= B*inv(I-(1-B)*W)
+
+        consoleVect <- "calculating matrix F"
+        if (!silent) {
+            print(consoleVect)
+        }
+
+        idMat <- diag(nrow = sizeVerts, ncol = sizeVerts)
+        preInvert <- idMat-(1-betaVal)*W #perform initial operations
+        postInvert <- solve(preInvert) #invert matrix
+        matrixF <- betaVal*postInvert #Finish and make matrix F
+
+    #======== Part 4 of Leis: calculating matrix E ===================
+
+        consoleVect <- c(consoleVect, "calculating matrix E")
+
+
+        #Make a diagonal matrix D, or vector of heat scores in diagonal form
+
+        D <- diag(x = AGGverts$Gene_Score)
+        dimnames(D) <- list(AGGverts$name,AGGverts$name)
+
+        #Calculate matrix E (weighted diffusion matrix) as F*D (F postmultiplied
+        #by D)
+
+        matrixE <- matrixF%*%D
+
+    #======== Part 5 of Leis: extracting 'heat' influence from matrix E =====
+
+        consoleVect <- c(consoleVect, "Assigning heat/influence scores from E to edges")
```

Comment:
You are writing all these progress messages - but then you are buffering them in ```consoleVect```. I don't understand why ... if you are not using them as progress feedback, you don't need to create them, because the fact that you passed these points is implied by the fact that you reach the bottom (when you do).